### DISCUSSION TITLE
Is it possible to implement constant propagation based on Slang AST?

### DISCUSSION THREAD

**Discussion by yanggeorge** [MikePopoloski/slang#1055] [General]
I just find some part of Spyglass lint rules depend this tech. 

Such as **DisabledAnd** Lint Rule is part of Spyglass, which is used to find disabled AND logic.

Consider the following example:
```systemverilog
module DisabledAnd(output x, output y);  
wire a,b,c,d,common;

assign x = a & b; //<-- 3. x is always disabled because of a
assign y = c & d;

assign a = common; // <-- 2. a is always 1'b0
assign c = common;

assign common = 1'b0; // <-- 1. common is always 1'b0
endmodule
```
For the above example, the following violation is reported:Input pin in1 on comb-and gate DisabledAnd.x (master RTL\_AND) is tiedlow(connected to DisabledAnd.a)

So is it possible to implement such based on current Slang AST ?  I just learns some about constant propagtion tech. I don't know if this appoach is correct.

**Comment by MikePopoloski** [comment]
Yes, it is possible. slang already does constant propagation in subexpressions, though it does not propagate constants through nets and variables, which is more challenging to do but still possible with some work.

**Reply by yanggeorge** [reply]
Thanks.
